libnmstate/nm/applier.py

```diff
@@ -170,17 +169,17 @@ def set_ifaces_admin_state(ifaces_desired_state, con_profiles=()):
                 InterfaceState.ABSENT,
             ):
                 nmdevs = _get_affected_devices(iface_desired_state)
-                for nmdev in nmdevs:
-                    remove_devs_actions[nmdev] = [
+                for affected_nmdev in nmdevs:
+                    remove_devs_actions[affected_nmdev] = [
                         device.deactivate,
                         device.delete,
                     ]
-                    if nmdev.get_device_type() in (
-                        nmclient.NM.DeviceType.OVS_BRIDGE,
-                        nmclient.NM.DeviceType.OVS_PORT,
-                        nmclient.NM.DeviceType.OVS_INTERFACE,
-                    ):
-                        remove_devs_actions[nmdev].append(device.delete_device)
+                if (
+                    nmdev.is_software()
+                    and iface_desired_state[Interface.STATE]
+                    == InterfaceState.ABSENT
```

Comment:
In todays standup @EdDev @cathay4t and me agreed to add an exception for veth interfaces here for now so that the test `test_removing_a_non_removable_iface` will not fail.

Reply:
Done, but I don't like the fact that it is not consistent.